activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -498,6 +498,11 @@ def within_new_transaction(isolation: nil, joinable: true)
             raise
           ensure
             unless error
+              # In 7.1 we enforce timeout >= 0.4.0 which now longer use throw, so we can
+              # go back to the original behavior of committing on non-local return.
+              # If users are using throw, we assume it's not an error case.
+              completed = true unless ActiveRecord.rollback_transaction_on_non_local_return
+
```

Comment:
There's a current deprecation down here for a case that was missed in the original 6.1 warning (#44607).

Should we immediately reverse that? It's bad and confusing to have two different behaviours depending on transaction nesting, but it also seems confusing to keep a deprecation pushing in the "wrong" direction.